apps/s_client.c

```diff
@@ -2980,10 +3079,17 @@ int s_client_main(int argc, char **argv)
             }
         }
 
-        if (SSL_is_dtls(con) && DTLSv1_handle_timeout(con) > 0)
-            BIO_printf(bio_err, "TIMEOUT occurred\n");
+        if (timeoutp != NULL) {
+            SSL_tick(con);
```

Comment:
SSL_tick seems poorly named ...

Reply:
The SSL_tick() name was first proposed and discussed a long time ago. I'm personally fine with the name. But if we want to change it then we should probably raise a separate issue for it and discuss it in OTC. I don't think it should hold up this PR.

@t-j-h - should I raise an issue for this?

Reply:
Let's get an OTC decision on the API related items so we have clear direction and can apply whatever the agreed principles end up being. Seeing it in use should help crystalise the views rapidly.

Reply:
I've put an OTC hold on the PR to discuss it.

Reply:
Marking this as resolved as it is sorted in https://github.com/openssl/openssl/issues/20867.